flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1277,6 +1284,38 @@ private void discardCheckpoint(PendingCheckpoint pendingCheckpoint, @Nullable Th
 		}
 	}
 
+	private void tryHandleCheckpointException(DeclineCheckpoint message, PendingCheckpoint currentCheckpoint) {
+		if (message.getReason() != null
+			&& !(message.getReason() instanceof CheckpointDeclineException)
+			&& failOnCheckpointingErrors
+			&& !currentCheckpoint.getProps().isSavepoint()) {
+
+			boolean processed = false;
+			ExecutionAttemptID failedExecutionID = message.getTaskExecutionId();
+			for (ExecutionVertex vertex : tasksToWaitFor) {
+				if (vertex.getCurrentExecutionAttempt().getAttemptId().equals(failedExecutionID)) {
+					if (currentPeriodicTrigger != null) {
+						currentPeriodicTrigger.cancel(true);
+						currentPeriodicTrigger = null;
+					}
+
+					vertex.fail(message.getReason());
+					processed = true;
+					break;
+				}
+			}
+
+			if (!processed) {
+				LOG.error("Can not find the failed execution id {} for job {} with checkpoint id {} " +
+						"in waiting acknowledge task list.",
+					failedExecutionID, message.getJob(), message.getCheckpointId());
+				throw new IllegalStateException("Can not find the failed execution id : " + failedExecutionID
+					+ " for job " + message.getJob() + " with checkpoint id" + message.getCheckpointId()
```

Comment:
hi, it's better add a blank after `with checkpoint id`

Reply:
good catch, updated.